plugin/storage/cassandra/factory.go

```diff
@@ -101,7 +102,12 @@ func (f *Factory) CreateSpanReader() (spanstore.Reader, error) {
 
 // CreateSpanWriter implements storage.Factory
 func (f *Factory) CreateSpanWriter() (spanstore.Writer, error) {
-	return cSpanStore.NewSpanWriter(f.primarySession, f.Options.SpanStoreWriteCacheTTL, f.primaryMetricsFactory, f.logger), nil
+	writer := cSpanStore.NewSpanWriter(f.primarySession, f.Options.SpanStoreWriteCacheTTL, f.primaryMetricsFactory, f.logger)
+
+	return decorator.NewRateLimitedWriter(
+		writer,
+		f.Options.writesPerSecond,
```

Comment:
if f.Options.writesPerSecond == 0, do we need to wrap?

Reply:
Lol see what @p.r suggested earlier. We are wrapping with noop rate limiter in that case.

Reply:
if we have an if statement, I'd rather have it here and not add two more layers of indirection. Then in your rate limited writer, return an error if rate < 1.

Reply:
Oh man. I want to wait on @vprithvi's response because this is literally what I was wondering above.

> Good idea. I wasn't sure if a passthrough would hurt performance due to the indirection.

https://github.com/jaegertracing/jaeger/pull/1147#discussion_r229048135

Reply:
I have a preference towards having this logic be in the rate limiter because it means that it doesn't need to be in every storage implementation. However, it's not a strong preference.

Reply:
That's a separate concern though, the config and application of rate limiter is currently in the cassandra factory, but we could easily pull it up to the higher level (meta) factory (I just don't think we need to do it right now).  This keeps the rate limiter simpler - you can only instantiate it with a positive rate.

Reply:
I'm wondering when we care about performance of indirection, etc. But I'll go back to the original if statement here.